src/Compilers/Core/Portable/Collections/ArrayBuilderExtensions.cs

```diff
@@ -160,6 +161,18 @@ public static T Pop<T>(this ArrayBuilder<T> builder)
             return e;
         }
 
+        public static bool TryPop<T>(this ArrayBuilder<T> builder, [MaybeNullWhen(false)] out T result)
+        {
+            result = default;
```

Comment:
This should be moved to be just above `return false`. Doing it here is adding redundant writes.